src/AddContact.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+
+function AddContact(props) {
+  const [name, setName] = useState("")
+  const [image_url, setImageUrl] = useState("");
+  const [email, setEmail] = useState("");
+  const [phone_number, setPhoneNumber] = useState("")
+
+  // assign new contact a random 8 digit id
+  const newId = Math.round(Math.random() * 100000000);
+
+  const handleSubmitChangeButton = () => {
+    props.addContact({
+      newId,
+      name,
+      image_url,
+      email,
+      phone_number,
+    })
+
+    // go back to the /contacts/ route (ContactList) when done
+    props.history.push('/contacts/')
+  }
+
+  // form to input new contact, will be added using handleSubmitChangeButton function above
+  return (
+    <div className="AddContact">
+      <div className="row">
+        <div className="col-md-6 offset-md-3">
+          <form className="post-form">
+            <h3>Add a New Contact</h3>
+
+            <div className="form-group">
+              <label>Full Name</label>
+              <input
+                type="text"
+                className="form-control"
+                onChange={(event) => setName(event.target.value)}
+              />
+              <br />
+              <label>Email Address</label>
+              <input
+                type="text"
+                className="form-control"
+                onChange={(event) => setEmail(event.target.value)}
+              />
+              <br />
+              <label>Phone Number</label>
+              <input
+                type="text"
+                className="form-control"
+                onChange={(event) => setPhoneNumber(event.target.value)}
+              />
+              <br />
+              <label>Image URL</label>
+              <input
+                type="text"
+                className="form-control"
+                onChange={(event) => setImageUrl(event.target.value)}
+              />
+              <br />
+            </div>
+
+            <button
+              onClick={handleSubmitChangeButton}
+              type="button"
+              className="btn btn-primary add-contact-and-return"
+            >
+              Add Contact
+            </button>
+          </form>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+
+AddContact.propTypes = {
+      name: PropTypes.string.isRequired,
```

Comment:
Watch your indenting

Reply:
Getting this error in console: `Warning: Failed prop type: The prop `name` is marked as required in `AddContact`, but its value is `undefined``